### DISCUSSION TITLE
Serialization and cancellable effects

### DISCUSSION THREAD

**Discussion by bergkvist** [zerobias/effector#383] [Q&A] [answered]
### In Redux, you can persist and later reload the entire state of the application:
1. Serialize the entire store as JSON (this is very easy when you only have a single store)
2. Persist the JSON-string with localStorage
3. Later reload localStorage - and have your application look exactly like it did earlier

### Actions (in Redux) can be serialized/sent between different clients
This can be really useful when designing real-time applications. One client could broadcast an action to all the other clients.

### Complex side-effect logic with redux-saga or rxjs for redux
Redux has extension libraries that allow for doing things like cancelling async functions to avoid race conditions.

---

## How does effector match up on these points?

**Comment by YanLobat** [comment, marked as answer]
1) For localStorage there are at least two libraries to work with:
https://www.npmjs.com/package/effector-persist
https://github.com/yumauri/effector-storage
or you could link store with LS via watchers like
`$someStore.watch((actualValue) => localStorage.setItem('myTestKey', actualValue)`
2) Also effector provides you domains(essential thing for testing and SSR/SSG) - namespaces, in other words you could create every application store under that domain and later on [serialize](https://effector.now.sh/docs/api/effector/serialize) it or [hydrate](https://effector.now.sh/docs/api/effector/hydrate)
3) Side-effects are managed via effects, [example](https://github.com/zerobias/effector#more-examples) with cancelation

More examples to go: https://github.com/zerobias/effector#more-examples

**Reply by doasync** [reply]
Cancelable effects can also  be created using this lib:
https://github.com/doasync/fry-fx
